execution.py

```diff
@@ -375,16 +374,13 @@ def execute(self, prompt, prompt_id, extra_data={}, execute_outputs=[]):
                           broadcast=False)
             executed = set()
             output_node_id = None
-            to_execute = []
 
-            for node_id in list(execute_outputs):
-                to_execute += [(0, node_id)]
+            to_execute = set(execute_outputs)
 
             while len(to_execute) > 0:
                 #always execute the output that depends on the least amount of unexecuted nodes first
-                memo = {}
-                to_execute = sorted(list(map(lambda a: (len(recursive_will_execute(prompt, self.outputs, a[-1], memo)), a[-1]), to_execute)))
-                output_node_id = to_execute.pop(0)[-1]
+                output_node_id = min(to_execute, key=lambda a: len(recursive_will_execute(prompt, self.outputs, a, {})))
```

Comment:
The point of passing a memo to the recursive_will_execute function is so it only needs to calculate things for each node once.

Reply:
Fixed